src/context/SemanticSearchContext.tsx

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+
+import React, {createContext, useContext, useState, useMemo, useCallback} from 'react';
+import {useLazySearchArticlesQuery} from 'store/services/articleApi';
+import {SearchResult, SemanticSearchContextType} from 'types/semanticSearch';
+import {useAppDispatch, useAppSelector} from 'store/hooks';
+import {setSemanticSearchStatus, resetSemanticSearch} from 'store/SemanticSearchSlice';
+import i18n from 'util/i18n';
+
+const SemanticSearchContext = createContext<SemanticSearchContextType | undefined>(undefined);
+
+interface SemanticSearchProviderProps {
+  children: React.ReactNode;
+}
+
+export function SemanticSearchProvider({children}: SemanticSearchProviderProps) {
+  const [triggerSearch, {isLoading}] = useLazySearchArticlesQuery();
+  const [augmentedResults, setAugmentedResults] = useState<SearchResult[]>([]);
+  const dispatch = useAppDispatch();
+  const token = useAppSelector((state) => state.auth.token);
+
+  const searchStatus = useAppSelector((state) => state.semanticSearch.searchStatus);
+
+  const performSearch = useCallback(
+    (query: string) => {
+      if (!query || query.trim() === '') {
+        setAugmentedResults([]);
+        dispatch(setSemanticSearchStatus('idle'));
+        return;
+      }
+
+      dispatch(setSemanticSearchStatus('loading'));
+      triggerSearch({searchQuery: query, lang: i18n.language, token})
+        .unwrap()
+        .then((apiResponse) => {
+          const lowerCaseQuery = query.toLowerCase();
+          const queryTerms = lowerCaseQuery.split(' ').filter((term) => term.length > 2);
+
+          const newResults: SearchResult[] = apiResponse.results.map((item) => {
+            const relevanceScore = Math.round(item.similarity * 100);
+
+            let classification: 'direct' | 'high' | 'related';
+            if (relevanceScore >= 90) {
+              classification = 'direct';
+            } else if (relevanceScore >= 80) {
+              classification = 'high';
+            } else {
+              classification = 'related';
+            }
```

Comment:
Is there a reason why you multiply here?
You mentioned the similarity can be between [-1,1] or is it (-1,1)?
In this case you might also want to use the absolute of the similarity? (assuming that both -1 and 1 mean near perfect similarity)

```suggestion
            const relevanceScore = item.similarity;

            let classification: 'direct' | 'high' | 'related';
            if (relevanceScore >= 0.9) {
              classification = 'direct';
            } else if (relevanceScore >= 0.8) {
              classification = 'high';
            } else {
              classification = 'related';
            }
```

Reply:
I adjusted the code to deal with negative numbers… it will become “unrelated”, but I do not think this option will ever appear